### DISCUSSION TITLE
Mobile application authentication based on OAUTH

### DISCUSSION THREAD

**Discussion by quick691fr** [goauthentik/authentik#7926] [Q&A]
Hi,

I'm trying to make a mobile app authentication using authentik Oauth to use protected token based REST API.
I can make it worked using switching from mobile app to web browser for user authentication part and go back to mobile app with token after successfully auth but the process is kinda strange to see app switching from my app to browser on mobile. Is there a way to make oauth signin directly from app login screen without using browser and authentik landing page ?

Thanks

**Comment by BeryJu** [comment]
You should be able to embed a WKWebView (or whatever the android equivalent is) into your app to show the login

**Reply by quick691fr** [reply]
I checked with Oauth RFC and they push to use the default browser instead of a webview apparently for security purposes.
I can find a workaround if I can find a logout page on authentik because the current /flows/-/default/invalidation/ page is currently redirecting to the login page not a logout page saying like "You are now disconnected, you can now close this page" kinda like the application endSessionEndpoint but with complete logout from authentik session.
